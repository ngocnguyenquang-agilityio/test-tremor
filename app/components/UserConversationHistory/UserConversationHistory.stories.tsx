// Libs
import type { Meta, StoryObj } from "@storybook/react";

// Components
import UserConversationHistory from "./UserConversationHistory";
import { PROFILE_ITEM } from "@/mocks/profile";

const meta = {
  title: "Components/UserConversationHistory",
  tags: ["autodocs"],
  component: UserConversationHistory,
} as Meta<typeof UserConversationHistory>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: () => (
    <UserConversationHistory
      avatar={PROFILE_ITEM.avatar}
      name={PROFILE_ITEM.name}
      lastConversation={PROFILE_ITEM.lastConversation}
    />
  ),
};
