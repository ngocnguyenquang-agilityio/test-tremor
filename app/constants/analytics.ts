/* Constants for Analytics Page */
export const ANALYTICS_STATISTICAL_TYPE = {
  BOOKINGS: "Bookings",
  REVENUE: "Revenue",
  TODAY_USER: "Today's Users",
  FOLLOWERS: "Followers",
};

export const enum CHART_TYPE {
  SALE = "sale",
  PERFORMANCE = "performance",
}
