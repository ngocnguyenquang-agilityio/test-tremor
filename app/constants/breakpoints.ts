export const BREAKPOINTS = {
  MD: 768,
  LG: 1024,
};
