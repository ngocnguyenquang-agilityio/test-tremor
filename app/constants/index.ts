export * from "./routes";
export * from "./regex";
export * from "./messages";
export * from "./profile";
export * from "./commons";
export * from "./flagSrc";
export * from "./sections";
export * from "./images";
