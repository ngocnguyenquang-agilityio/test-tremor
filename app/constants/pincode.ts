const DEFAULT_PIN_CODE = "*";
const PIN_CODE_LENGTH = 6;

export { DEFAULT_PIN_CODE, PIN_CODE_LENGTH };
