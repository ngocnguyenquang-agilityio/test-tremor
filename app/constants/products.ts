// Types
import { SelectOptionData } from "@/types";

type TopSellingProductsType = {
  [key: number]: string;
  1: string;
  2: string;
  3: string;
  4: string;
  5: string;
};

export const TOP_SELLING_PRODUCTS_SRC: TopSellingProductsType = {
  1: "/assets/images/products/blue-shoe.webp",
  2: "/assets/images/products/black-mug.webp",
  3: "/assets/images/products/black-chair.webp",
  4: "/assets/images/products/bang-sound.webp",
  5: "/assets/images/products/photo-tools.webp",
};

export const TYPE_PRICE: SelectOptionData[] = [
  {
    option: "BTC",
    value: "1",
  },
  {
    option: "CNY",
    value: "2",
  },
  {
    option: "EUR",
    value: "3",
  },
  {
    option: "GBP",
    value: "4",
  },
  {
    option: "INR",
    value: "5",
  },
  {
    option: "USD",
    value: "6",
  },
];

export const TAGS_PRICE: SelectOptionData[] = [
  {
    option: "Black Friday",
    value: "1",
  },
  {
    option: "Expired",
    value: "2",
  },
  {
    option: "Out of Stock",
    value: "3",
  },
  {
    option: "In Stock",
    value: "4",
  },
  {
    option: "Sale",
    value: "5",
  },
];

export const CATEGORY_PRODUCT: SelectOptionData[] = [
  {
    option: "Clothing",
    value: "1",
  },
  {
    option: "Electronics",
    value: "2",
  },
  {
    option: "Furniture",
    value: "3",
  },
  {
    option: "Others",
    value: "4",
  },
  {
    option: "Real Estate",
    value: "5",
  },
];

export const SIZE_PRODUCT: SelectOptionData[] = [
  {
    option: "Extra Large",
    value: "1",
  },
  {
    option: "Extra Small",
    value: "2",
  },
  {
    option: "Large",
    value: "3",
  },
  {
    option: "Medium",
    value: "4",
  },
  {
    option: "Small",
    value: "5",
  },
];
