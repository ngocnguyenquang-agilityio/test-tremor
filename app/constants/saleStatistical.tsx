import { ReactNode } from "react";
import { IoMdPersonAdd } from "react-icons/io";
import { MdLeaderboard, MdStore, MdWeekend } from "react-icons/md";

interface Icon {
  bgIcon: string;
  icon: ReactNode;
}

interface Sale {
  [key: string]: Icon;
}

export const SALE_STATISTICAL: Sale = {
  bookings: {
    bgIcon: "bg-gradient-arsenic",
    icon: <MdWeekend color="white" size="24px" />,
  },
  todayUser: {
    bgIcon: "bg-[linear-gradient(195deg,#49a3f1,#1A73E8)]",
    icon: <MdLeaderboard color="white" size="24px" />,
  },
  revenue: {
    bgIcon: "bg-[linear-gradient(195deg,#66BB6A,#43A047)]",
    icon: <MdStore color="white" size="24px" />,
  },
  followers: {
    bgIcon: "bg-[linear-gradient(195deg,#EC407A,#D81B60)]",
    icon: <IoMdPersonAdd color="white" size="24px" />,
  },
};
