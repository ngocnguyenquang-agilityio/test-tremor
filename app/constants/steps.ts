import { Step } from "@/types";

export const NEW_PRODUCT_FORM_STEPS: Step[] = [
  { index: 1, content: "Product Info" },
  { index: 2, content: "Media" },
  { index: 3, content: "Social" },
  { index: 4, content: "Pricing" },
];
