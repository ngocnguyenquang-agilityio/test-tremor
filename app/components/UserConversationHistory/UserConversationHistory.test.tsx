import { render } from "@testing-library/react";
import UserConversationHistory from "./UserConversationHistory";

// Constants
import { AVATAR_SRC } from "@/constants";

describe("UserConversationHistory component", () => {
  const props = {
    avatar: AVATAR_SRC.md,
    name: "Sophie.B",
    lastConversation: "Hi! I need more information..",
  };

  const userConversationHistory = () => {
    return render(<UserConversationHistory {...props} />);
  };

  it("Should render UserConversationHistory snapshot correctly", () => {
    expect(userConversationHistory()).toMatchSnapshot();
  });
});
