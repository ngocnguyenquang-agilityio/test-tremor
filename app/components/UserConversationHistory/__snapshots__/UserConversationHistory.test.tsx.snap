// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserConversationHistory component Should render UserConversationHistory snapshot correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="tremor-Flex-root flex w-full flex-row justify-between items-center mb-2 py-2 flex-wrap xs:flex-nowrap"
      >
        <div
          class="tremor-Flex-root flex w-full flex-row items-center justify-start flex-wrap sm:flex-nowrap"
        >
          <div
            class="w-12 h-12"
          >
            <div
              class="inline-flex rounded-full overflow-hidden shadow-md"
            >
              <img
                alt="Sophie.B"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="48"
                src="http://localhost/_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=96&q=75"
                srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=48&q=75 1x, /_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=96&q=75 2x"
                style="color: transparent;"
                width="48"
              />
            </div>
          </div>
          <div
            class="max-w-[180px] font-medium dark:text-white pl-4"
          >
            <p
              class="text-tremor-default text-sm text-primary dark:text-dark-primary font-semibold"
            >
              Sophie.B
            </p>
            <p
              class="text-tremor-default text-xs text-secondary font-light truncate dark:text-dark-romance"
            >
              Hi! I need more information..
            </p>
          </div>
        </div>
        <button
          class="tremor-Button-root flex-shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default shadow-tremor-input dark:shadow-dark-tremor-input px-4 py-2 text-sm border-tremor-brand dark:border-dark-tremor-brand text-tremor-brand-inverted dark:text-dark-tremor-brand-inverted hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis min-w-[64px] m-0 border-0 hover:bg-[none] bg-transparent dark:bg-transparent dark:hover:bg-[none]"
        >
          <span
            class="tremor-Button-text text-tremor-default whitespace-nowrap"
          >
            <p
              class="text-tremor-default dark:text-dark-tremor-content text-primary text-xs text-center font-bold uppercase leading-[17px]"
            >
              Reply
            </p>
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="tremor-Flex-root flex w-full flex-row justify-between items-center mb-2 py-2 flex-wrap xs:flex-nowrap"
    >
      <div
        class="tremor-Flex-root flex w-full flex-row items-center justify-start flex-wrap sm:flex-nowrap"
      >
        <div
          class="w-12 h-12"
        >
          <div
            class="inline-flex rounded-full overflow-hidden shadow-md"
          >
            <img
              alt="Sophie.B"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="48"
              src="http://localhost/_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=96&q=75"
              srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=48&q=75 1x, /_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=96&q=75 2x"
              style="color: transparent;"
              width="48"
            />
          </div>
        </div>
        <div
          class="max-w-[180px] font-medium dark:text-white pl-4"
        >
          <p
            class="text-tremor-default text-sm text-primary dark:text-dark-primary font-semibold"
          >
            Sophie.B
          </p>
          <p
            class="text-tremor-default text-xs text-secondary font-light truncate dark:text-dark-romance"
          >
            Hi! I need more information..
          </p>
        </div>
      </div>
      <button
        class="tremor-Button-root flex-shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default shadow-tremor-input dark:shadow-dark-tremor-input px-4 py-2 text-sm border-tremor-brand dark:border-dark-tremor-brand text-tremor-brand-inverted dark:text-dark-tremor-brand-inverted hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis min-w-[64px] m-0 border-0 hover:bg-[none] bg-transparent dark:bg-transparent dark:hover:bg-[none]"
      >
        <span
          class="tremor-Button-text text-tremor-default whitespace-nowrap"
        >
          <p
            class="text-tremor-default dark:text-dark-tremor-content text-primary text-xs text-center font-bold uppercase leading-[17px]"
          >
            Reply
          </p>
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
